### ISSUE TITLE
FirstHashElementLineBreak is reading keyword arguments like hashes

### ISSUE THREAD

**Issue by jfelchner** [bbatsov/rubocop#2554]
`FirstHashElementLineBreak` shouldn't deal with arguments in a method.

```ruby
my_method_call(with_keyword: 'blah',
               args: 'blargh')
```

Throws a `FirstHashElementLineBreak` error because it assumes it should be:

```ruby
my_method_call(
  with_keyword: 'blah',
  args: 'blargh')
```

which is obviously incorrect.  Whether there's a line break there should be handled by `FirstMethodParameterLineBreak` instead.

**Comment by alexdowad** [comment]
`parser` parses those keyword arguments as a hash. I suspect that the MRI parser likely does the same.

Those arguments could very well *be* a hash, if `my_method_call` was not declared to take KW arguments.

I haven't studied the implementation, but I strongly suspect that the Ruby interpreter implements KW arguments by packing the arguments up in a hash and then automatically unpacking them for you.

Having said all that, you do seem to have a point here. If there were explicit curly braces around those keywords, I think that `FirstHashElementLineBreak` would be right here. But without the explicit braces... maintainers, what do you think?

**Comment by jonas054** [comment]
Yes. When there are no curly braces, I think it makes sense to move the responsibility from `FirstHashElementLineBreak` to `FirstMethodArgumentLineBreak`.

**Comment by alexdowad** [comment]
What about:

```ruby
method("not a keyword" => "value",
  "another" => "pair")
```

Is this handled by `FirstHashElementLineBreak`, or `FirstMethodArgumentLineBreak`?

**Comment by jfelchner** [comment]
IMO `FirstMethodArgumentLineBreak`. I think the fact that it's a method argument is a "more special case" and therefore should supersede the fact that it's a hash. 

I mean, what we should probably do is give both cops more granular options, but that's a lot more work.

**Comment by alexdowad** [comment]
Pushed a fix for this.

**Comment by jfelchner** [comment]
@alexdowad that looks spectacular :grinning:
